### PR TITLE
update ScientificTypes to ScientificTypesBase

### PR DESCRIPTION
**Pull request by OkonSamuel**
In line with the migration of https://github.com/alan-turing-institute/ScientficTypes.jl` to https://github.com/JuliaAI/ScientificTypesBase.jl and https://github.com/JuliaAI/MLJScientificTypes.jl to https://github.com/JuliaAI/ScientificTypes.jl, this PR should be merged before https://github.com/JuliaAI/ScientificTypes.jl/pull/126 gets merged. 
cc. @ablaom  cc. @juliohm

### CODE REVIEW

**Comment by ablaom** [comment]
@OkonSamuel Yes, good catch, thanks!

File: Project.toml [hunk 1]
```diff
@@ -1,18 +1,32 @@
-name = "StatisticalTraits"
-uuid = "64bff920-2084-43da-a3e6-9bb72801c0c9"
+name = "ScientificTypes"
```
Comment (ablaom) [inline Project.toml:1]: ???

Surely you don't mean to make all these changes to the Project.toml (including changing the name of the StatisticalTraits package)? You just want to replace one of the deps.

Comment (OkonSamuel) [reply Project.toml]: Yeah. An error on my part.

**Comment by ablaom** [comment]
@OkonSamuel 

We need to fix tests as well: https://github.com/alan-turing-institute/StatisticalTraits.jl/blob/b5c8be1bc76bbbea7a97989398434847804d3f10/test/runtests.jl#L3

**Comment by OkonSamuel** [comment]
> @OkonSamuel
> 
> We need to fix tests as well:
> 
> https://github.com/alan-turing-institute/StatisticalTraits.jl/blob/b5c8be1bc76bbbea7a97989398434847804d3f10/test/runtests.jl#L3
Yes

**Pull request merged**